### PR TITLE
More docs.

### PR DESCRIPTION
**Pull request by Hixie**
cc @abarth

### CODE REVIEW

**Comment by Hixie** [comment]
Fixes https://github.com/flutter/flutter/issues/10165, among other things.

File: packages/flutter/lib/src/widgets/basic.dart [hunk 1]
```diff
@@ -1171,8 +1171,18 @@ class FractionallySizedBox extends SingleChildRenderObjectWidget {
 /// A box that limits its size only when it's unconstrained.
 ///
 /// If this widget's maximum width is unconstrained then its child's width is
-/// limited to maxWidth. Similarly, if this widget's maximum height is unconstrained
-/// then its child's height is limited to to maxHeight.
+/// limited to [maxWidth]. Similarly, if this widget's maximum height is
+/// unconstrained then its child's height is limited to to [maxHeight].
```
Comment (mravn-google) [inline packages/flutter/lib/src/widgets/basic.dart]: Duplicate `to`

File: packages/flutter/lib/src/widgets/container.dart [hunk 2]
```diff
@@ -123,16 +123,37 @@ class DecoratedBox extends SingleChildRenderObjectWidget {
...
+/// widget has its own opinions regarding the size that the [Container] should
+/// take):
+///
+/// ```dart
+/// new Center(
+///   child: new Container(
+///     color: const Color(0xFF00FF00),
+///     width: 48.0,
+///     height: 48.0,
+///   ),
```
Comment (mravn-google) [inline packages/flutter/lib/src/widgets/container.dart]: Should we not use the `Container`'s `child` property in the code sample?

File: packages/flutter/lib/src/widgets/editable_text.dart [hunk 3]
```diff
@@ -61,6 +61,10 @@ class TextEditingController extends ValueNotifier<TextEditingValue> {
 
   /// The current string the user is editing.
   String get text => value.text;
+  /// Setting this will notify all the consumers of this [TextEditingController]
+  /// that they need to update (it calls [notifyListeners]). For this reason,
```
Comment (mravn-google) [inline packages/flutter/lib/src/widgets/editable_text.dart]: Can we avoid talking about both `consumers` and `listeners`?

**Review by mravn-google — APPROVED** [review]
LGTM

File: packages/flutter/lib/src/widgets/basic.dart (hunk 1 above)
Comment (Hixie) [inline packages/flutter/lib/src/widgets/basic.dart]: fixed

File: packages/flutter/lib/src/widgets/editable_text.dart (hunk 3 above)
Comment (Hixie) [inline packages/flutter/lib/src/widgets/editable_text.dart]: Changed "consumers" to "listeners".

File: packages/flutter/lib/src/widgets/container.dart (hunk 2 above)
Comment (Hixie) [inline packages/flutter/lib/src/widgets/container.dart]: Fixed.

**Comment by Hixie** [comment]
will land on green

**Pull request merged**